src/nix/util.rs

```diff
@@ -1,14 +1,18 @@
-/// Build a shell command that the user can run as-is.
-pub fn build_shell_command(program: String, args: &Vec<String>) -> String {
-    format!(
-        "{} {}",
-        program,
-        args.iter()
-            .map(|x|
-                // If the argument contains a special character, it must be
-                // quoted. We check only for "?", though.
-                if x.contains("?") { format!("\"{}\"", x) } else { x.to_string() })
```

Comment:
Why was this behaviour (conditional quoting) removed? At least, I don't see it in `print_shell_command`.

Reply:
Oh that's my bad, I meant to keep it in but I must have lost it in between rounds of iterating, I will bring it back!

Reply:
Aside: we should add tests for this function (to catch regressions like this in the future).